substrate/frame/support/src/traits/tokens/imbalance/on_unbalanced.rs

```diff
@@ -53,3 +55,29 @@ pub trait OnUnbalanced<Imbalance: TryDrop> {
 }
 
 impl<Imbalance: TryDrop> OnUnbalanced<Imbalance> for () {}
+
+/// Resolves received asset credit to account `A`, implementing [`OnUnbalanced`].
+///
+/// Credits that cannot be resolved to account `A` are dropped. This may occur if the account for
+/// address `A` does not exist and the existential deposit requirement is not met.
+pub struct ResolveTo<A, F>(PhantomData<(A, F)>);
+impl<A: TypedGet, F: fungible::Balanced<A::Type>> OnUnbalanced<fungible::Credit<A::Type, F>>
+	for ResolveTo<A, F>
+{
+	fn on_unbalanced(credit: fungible::Credit<A::Type, F>) {
```

Comment:
The docs on `on_nonzero_unbalanced` say: "You probably want to implement this rather than `on_unbalanced`." What is the motivation for `on_unbalanced`? In the use case you have in mind it could be zero?

Reply:
Good catch. It's because it will try to drop it if zero first, and only after to resolve it if non zero.
I will update this impl.

Reply:
updated. also update the description with the motivation